addons/l10n_ma/models/res_company.py

```diff
@@ -0,0 +1,31 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from markupsafe import Markup
+
+from odoo import api, fields, models
+
+from odoo.addons.base.models.ir_qweb_fields import nl2br
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    ice = fields.Char("ICE", help='Common Company Identifier', default='')
+
+class BaseDocumentLayout(models.TransientModel):
+    _inherit = 'base.document.layout'
+
+    @api.model
+    def _default_company_details(self):
+        company = self.env.company
+        address_format, company_data = company.partner_id._prepare_display_address()
+        address_format = self._clean_address_format(address_format, company_data)
+        # company_name may *still* be missing from prepared address in case commercial_company_name is falsy
+        if 'company_name' not in address_format:
+            address_format = '%(company_name)s\n' + address_format
+            company_data['company_name'] = company_data['company_name'] or company.name
+        if company.country_code == 'MA':
+            address_format += '\nICE: %(ice)s\n'
+            company_data['ice'] = company.ice
+        return Markup(nl2br(address_format)) % company_data
```

Comment:
I didn't check but it's probably not correct: I suppose you want to inherit an existing field and add the new ICE field in the existing behavior. Here you're just copy/pasting the whole logic defined regardless of this behavior was already inherited somewhere. That just doesn't compose with potential other modules trying to modify this behavior.

So, what I'm expecting is a call to super() and an if statement that will add the ICE field in case the company is morocan.